### PR TITLE
feat: Search by city

### PR DESCRIPTION
**Pull request by andy-liubinski-stefanini**
-Weather app is now functional.

Only history of searches remains to be implemented.

### CODE REVIEW

**Comment by vadimb** [comment]
Try to use https://www.conventionalcommits.org/en/v1.0.0/ for naming

File: src/weatherFunctions/assembleAllWeather.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import getTodayWeather from './getTodayWeather';
+import getFutureWeather from './getFutureWeather';
+
+const assembleAllWeather = async location => {
```
Comment (vadimb) [inline src/weatherFunctions/assembleAllWeather.jsx:4]: WeatherService.js

File: src/weatherFunctions/assembleAllWeather.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  try {
+    const todayWeather = await getTodayWeather(location);
+    const forecastWeather = await getFutureWeather(location);
+
+    return [todayWeather, forecastWeather];
+  } catch (error) {
+    console.error(
+      'Cannot fetch weather data. Better go look out of the window.'
+    );
```
Comment (vadimb) [inline src/weatherFunctions/assembleAllWeather.jsx:13]: Dont log in service

File: src/components/CitySection.jsx [hunk 3]
```diff
@@ -7,12 +7,14 @@ const CitySection = () => {
...
-  const cityCountryString = `${weatherData?.cityName}, ${weatherData?.countryCode}`;
+  const weatherObject = weatherData && weatherData[0] ? weatherData[0] : null;
+  const cityCountryString = weatherObject
+    ? `${weatherObject.cityName}, ${weatherObject.countryCode}`
+    : 'Loading...';
   const todayString =
     datesArray.length > 0
-      ? `${datesArray[0]['formattedWeekday']}, ${datesArray[0]['formattedDate']}`
+      ? `${datesArray[1]['formattedWeekday']}, ${datesArray[1]['formattedDate']}`
       : '';
```
Comment (vadimb) [inline src/components/CitySection.jsx:17]: useWeatherByCity()

File: src/components/CitySection.jsx [hunk 4]
```diff
@@ -7,12 +7,14 @@ const CitySection = () => {
   const { searchFieldVisible, weatherData, datesArray } =
     useContext(AppContext);
 
-  const cityCountryString = `${weatherData?.cityName}, ${weatherData?.countryCode}`;
+  const weatherObject = weatherData && weatherData[0] ? weatherData[0] : null;
+  const cityCountryString = weatherObject
+    ? `${weatherObject.cityName}, ${weatherObject.countryCode}`
+    : 'Loading...';
   const todayString =
```
Comment (vadimb) [inline src/components/CitySection.jsx:14]: remove String suffix

File: src/components/CitySection.jsx [hunk 5]
```diff
@@ -7,12 +7,14 @@ const CitySection = () => {
   const { searchFieldVisible, weatherData, datesArray } =
     useContext(AppContext);
 
-  const cityCountryString = `${weatherData?.cityName}, ${weatherData?.countryCode}`;
+  const weatherObject = weatherData && weatherData[0] ? weatherData[0] : null;
```
Comment (vadimb) [inline src/components/CitySection.jsx:10]: looks abit wierd shoydl the service return emry array

File: src/components/ForecastBox.jsx [hunk 6]
```diff
@@ -1,13 +1,16 @@
 import ForecastDay from './ForecastDay';
+import { AppContext } from '../store/AppContext';
+import { useContext } from 'react';
 
 const ForecastBox = () => {
+  const { weatherData, datesArray } = useContext(AppContext);
+  const forecastArray = weatherData && weatherData[1] ? weatherData[1] : null;
```
Comment (vadimb) [inline src/components/ForecastBox.jsx:7]: review ...

File: src/components/ForecastDay.jsx [hunk 7]
```diff
@@ -1,14 +1,40 @@
...
+    ? Math.floor((9 / 5) * Number(maxTempC) + 32)
+    : null;
+
+  const minTempF = minTempC
+    ? Math.floor((9 / 5) * Number(minTempC) + 32)
+    : null;
+
+  const iconLink = day
+    ? `https://cdn.weatherbit.io/static/img/icons/${day.icon}.png`
+    : null;
```
Comment (vadimb) [inline src/components/ForecastDay.jsx:21]: encapsulate in hook

File: src/components/CitySection.jsx (hunk 4 above)
Comment (andy-liubinski-stefanini) [reply src/components/CitySection.jsx:14]: done for String and Object

**Pull request merged**